app/controllers/admin/exports_controller.rb

```diff
@@ -0,0 +1,42 @@
+module Admin
+  class ExportsController < BaseController
+
+    def completions
+      respond_to do |format|
+        format.html { redirect_to admin_dashboard_index_path }
+        format.csv { send_data Export.to_csv_for_completion_report(data_for_completions_report) }
+      end
+    end
+
+    def data_for_completions_report
+      grouped = {}
+      current_site = Organization.find_by(subdomain: request.subdomain)
+      zip_codes = CourseProgress.where.not(completed_at: nil).joins(:user).merge(User.with_role(:user, current_site))
+                                                             .map { |p| p.user.profile.zip_code }
```

Comment:
Looking at this, I can see how complex of a query needed to be done to get this info. I'm curious if a .includes(:profile) (or .joins(:profile)) could be done on the user table. Then you could do a .pluck(:zip_code) perhaps? I'm not sure, as I'm sure it would take some finagling, so don't let it hold you up on merging this, but I think that this might be something to watch as Users/CourseProgresses grow in number. (Maps are often a source of performance pain.)